Draft-Accepted/RFCNNNN-Install-PowerShell.md

```diff
@@ -0,0 +1,218 @@
+some customers due to discoverability and some that have policies that don't allow for installing
+software onto Windows after initial provisioning.
+
+To compensate for these situations, customers will implement and use the "tools at hand" to perform
+management tasks. This leaves **Windows PowerShell 5.1**, not **PowerShell 7**, as the only
+automation solution available to many of them. While customers are missing out on useful new
+features that they may wish to have, they are certainly not receiving the performance and security
+improvements that come with each new release.
+
+This specification is an effort to help IT admins that wish to use new and updated releases of
+PowerShell on Windows and to easily install PowerShell from **Windows PowerShell 5.1** with a
+cmdlet. While this may not help customers that have post-deployment policy restrictions, it will
+help the majority of customers discover and update to the latest version of PowerShell for Windows.
+
+```
+As an admin,
+I can use Windows PowerShell 5.1 to provision multiple hosts with PowerShell 7 over PSRemoting,
+so that I can leverage new capabilities in PS7.
+```
+
+```
+As a user,
+I can discover and install PowerShell 7 from Windows PowerShell 5.1 on my Windows machine,
+so that I can leverage new capabilities in PS7.
+```
+
+## Shipping in Windows
+
+There are several issues that prevent shipping PowerShell 7 in Windows;
+
+- Size
+- Support lifecycle differences
+
+### Size constraints
+
+ A complete installation of PowerShell 7 currently is around 200 MB (unpacked). This is due to
+ including everything (modules and .NET assemblies) to provide parity with what Windows PowerShell
+ 5.1 with .NET Framework provides. There is currently no plan by Windows or the .NET team to ship
+ newer .NET in Windows, so PS7 will need to continue to ship .NET with PS7.
+
+### Support lifecycle
+
+ Windows has a 5+5 year support lifecycle. .NET supports up to 3 years for Long Term Support (LTS)
+ versions. So even if size was not an issue, we cannot check in a version of PS7 with a .NET that is
+ not supported while Windows itself is supported.
+
+### Solution
+
+To solve these issues, the proposal is to ship a new cmdlet in Windows and the PowerShell Gallery
+that would;
+
+- Be discoverable using normal PowerShell conventions
+- Perform a simple download and install of the latest stable release of PowerShell
+- Provide advanced options with parameters;
+  - Specific version
+  - Stable, Preview or LTS
+  - Scope - location of installation based on permissions
+
+## Goals/Non-Goals
+
+Goals:
+
+- Be discoverable as other built-in cmdlets
+- By default, download and install the latest stable release to the users CurrentUser Scope
+  - CurrentUser - default installation scope
+  - AllUsers - Administrative users may explicit select this scope
+- Allow users to use remoting for install at scale with Invoke-Command
```

Comment:
The ConfirmImpact == High may prevent unattended use (need to check).  May need a `-Force`.